src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,180 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import core.basesyntax.service.InvalidDataException;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceTest {
+    private static final int ADULT_AGE = 18;
+    private static final int ANOTHER_ADULT_AGE = 25;
+    private static final int NEGATIVE_AGE = -10;
+    private static final int TEEN_AGE = 17;
+    private static final int ZERO_AGE = 0;
+
+    private static RegistrationService registrationService;
+    private User john;
+    private User jane;
+    private User maria;
+    private User george;
+    private String validPassword;
+    private String anotherValidPassword;
+    private String invalidPassword;
+    private String shortPassword;
+    private String validLogin;
+    private String anotherValidLogin;
+    private String thirdValidLogin;
+    private String invalidLogin;
+    private String shortLogin;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
```

Comment:
Consider using `@BeforeEach` instead of `@BeforeAll` for initializing `RegistrationService`. `@BeforeAll` is more suitable for static initialization that is shared across all tests, while `@BeforeEach` is better for per-test setup where each test should start with a fresh instance of the class.